models/quickbooks__profit_and_loss.sql

```diff
@@ -0,0 +1,30 @@
+with general_ledger_by_period as (
+    select *
+    from {{ref('quickbooks__general_ledger_by_period')}}
+    where financial_statement_helper = 'income_statement'
+
+), final as (
+    select
+        {% if target.type == 'bigquery' %}
```

Comment:
same questions as balance sheet, why cast as strings?

Reply:
Reverted and changed to the `period_first_day as calendar_date` to show year and month within the single field opposed to the previous unnecessary logic.